### DISCUSSION TITLE
File upload with response

### DISCUSSION THREAD

**Discussion by jasonbdt** [inertiajs/inertia#487] [Help] [answered]
Hello,

I am currently trying to upload files via Inertia.js / Laravel in combination with Vue2.

The file upload itself already works using this:

```js
this.$inertia.post('/file-upload', formData, {
    ...options
});
```

But I am not able to get a response about: was file upload successfully or did errors occured?

As mentioned in inertia.js-Documentation (here) I should use `return Inertia::render('Component', ...props)` but if I do, the history will be updated to the URL which is a post request and not what I want. Do I miss something or is this not possible?

Dear regards
Jason

**Comment by jasonbdt** [comment]
Okey, I see my question is a duplicate of https://github.com/inertiajs/inertia/discussions/453 and there is already a Pull Request working on it https://github.com/inertiajs/inertia-laravel/pull/206. This leads me to the next question: when this will be merged and usable for all?

Changed the file in core for now but you know... it's not recommended

**Reply by claudiodekker** [reply]
I figured I'd address this separately here, as otherwise it would look as if I/we're purposefully ignoring that PR, which is not the case. The long story short is that **I don't have an answer to this question**: We don't know _if_ it will be merged at all _and_ when, or whether we'll instead end up providing a better solution altogether at some point.

The problem with any PR that changes the way that Inertia works, is that they take quite some time to review it, as well as to consider: 
- What are the use-cases? 
- What impact does it have?
- Is there a real need for this, or is there possibly a better way to solve (e.g.) the underlying problem?
- How does it affect our vision for Inertia's future?
- etc.

In the past, this process easily took us multiple hours for difficult PR's, to sometimes even _days_ when we did come up with a better long-term solution, and even then we sometimes decided to postpone ideas/discard solutions.

Anyway, I hope this explains why we haven't been getting around to PR's but have been answering relatively simple questions such as these, as they only take ~10 minutes or so. I hope we can give Inertia some more code-related love soon-ish, but both @reinink and I have been relatively packed with other things recently.

**Comment by claudiodekker** [comment, marked as answer]
Alright, so, usually the straight-forward solution here is to use the same URL for uploads, in which case the URL's will remain the same and the history doesn't get updated.

However, I can totally imagine situations in which this isn't the case for whatever reason, and in those situations this is what I would do:
1. Make an upload request (as you're already doing).
2. Flash the success state to the session.
3. Use `Redirect::back()` to go 'back' to the page you're already on. [This will prevent the URL from being updated, as well as the history state from being changed](https://inertiajs.com/manual-visits#browser-history).
4. In the `Inertia::render()` call (or in the global middleware), grab the state from the session (step 2) and render it.
5. Use your view to access this success state.

**Reply by jasonbdt** [reply]
Thanks for your advice, this works for me too!

P.s.: I never wanted to say you're not working on this / ignoring PR's. It was just a solution for me which worked and I hadn't any idea how to do it other way. Still loving this project and working with it :)
